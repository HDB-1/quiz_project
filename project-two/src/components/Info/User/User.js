import React from 'react'

export default function User(props) {
    return (
        <div>
            <p id="currentUser">Player {props.currentUser + 1}</p>
        </div>
    )
}
