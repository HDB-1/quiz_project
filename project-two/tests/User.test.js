import React from 'react';
import { shallow , mount } from 'enzyme';
import User from '../src/components/Info/User/User'



describe('Shallow User', () => {
  let wrapper;
  
  beforeEach(() => wrapper = shallow(<User currentUser={1}/>)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render one currentUser tag", () => {
      expect(wrapper.find('#currentUser').length).toEqual(1);
  });
  it("Should render the correct User", () => {
    expect(wrapper.find('#currentUser').text()).toEqual("Player 1");
  });
});
