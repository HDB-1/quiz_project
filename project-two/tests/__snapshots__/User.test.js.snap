// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow User Should render correctly 1`] = `
<div>
  <p
    id="currentUser"
  >
    Player 
    1
  </p>
</div>
`;
